From 1edb67344cf567237ad8092494b593b66c1c7d8b Mon Sep 17 00:00:00 2001
From: Colin Guthrie <colin@mageia.org>
Date: Tue, 16 Oct 2012 11:50:41 +0100
Subject: [PATCH 510/510] journal: Set the last_unused pointer correctly when
 attaching an unused window

It seems the previous code was copy/pasted from context_detach_window()
but not updated.
---
 src/journal/mmap-cache.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/journal/mmap-cache.c b/src/journal/mmap-cache.c
index 7813f03..251aefe 100644
--- a/src/journal/mmap-cache.c
+++ b/src/journal/mmap-cache.c
@@ -205,8 +205,8 @@ static void context_attach_window(Context *c, Window *w) {
         if (w->in_unused) {
                 /* Used again? */
                 LIST_REMOVE(Window, unused, c->cache->unused, w);
-                if (!c->cache->last_unused)
-                        c->cache->last_unused = w;
+                if (c->cache->last_unused == w)
+                        c->cache->last_unused = w->unused_prev;
 
                 w->in_unused = false;
         }
-- 
1.7.12.3

